### PR TITLE
PUT and DELETE character

### PR DESCRIPTION
**Pull request by aghigo**
Implements PUT and DELETE character

### CODE REVIEW

File: src/main/java/com/javachallengers/simpson/model/SimpsonCharacter.java [hunk 1]
```diff
@@ -2,11 +2,19 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
 @Document(collection = "simpson-character")
 public class SimpsonCharacter implements Serializable {
```
Comment (rafadelnero) [inline src/main/java/com/javachallengers/simpson/model/SimpsonCharacter.java:19]: Do we need to open getters and setters and a builder?

The ideal is to isolate our fields as much as possible. Can we change from @Data to @Getter?

File: src/main/java/com/javachallengers/simpson/model/dto/ApiErrorDTO.java [hunk 2]
```diff
@@ -3,35 +3,16 @@
...
 
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
 public final class ApiErrorDTO implements Serializable {
 	private static final long serialVersionUID = 5160147982546412023L;
 
 	private final String error;
```
Comment (rafadelnero) [inline src/main/java/com/javachallengers/simpson/model/dto/ApiErrorDTO.java:14]: Do we need to expose all our fields here?

File: src/main/java/com/javachallengers/simpson/model/dto/SimpsonCharacterRequestDTO.java [hunk 3]
```diff
@@ -7,55 +7,27 @@
 import javax.validation.constraints.NotNull;
 import javax.validation.constraints.Past;
 
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
 public final class SimpsonCharacterRequestDTO implements Serializable {
```
Comment (rafadelnero) [inline src/main/java/com/javachallengers/simpson/model/dto/SimpsonCharacterRequestDTO.java:15]: Do we need to expose all our fields?

File: src/main/java/com/javachallengers/simpson/service/SimpsonCharacterService.java [hunk 4]
```diff
@@ -30,17 +33,58 @@ public SimpsonCharacterService(SimpsonCharacterRepository simpsonCharacterReposi
...
+		if(currentCharacter.isEmpty()) {
+			throw new SimpsonCharacterException("Character not found");
+		}
+		
+		Optional<SimpsonCharacter> anotherCharacter = simpsonCharacterRepository.findByNameAndSurname(data.getName(), data.getSurname());
+		if(anotherCharacter.isPresent() && !anotherCharacter.get().getId().equals(id)) {
+			throw new SimpsonCharacterException("Another character already exists with the same name " + data.getName() + " and surname " + data.getSurname());
+		}
+	
+		SimpsonCharacter character = currentCharacter.get();
```
Comment (rafadelnero) [inline src/main/java/com/javachallengers/simpson/service/SimpsonCharacterService.java:67]: Can we create a converter class that encapsulates the object conversion?

File: src/main/java/com/javachallengers/simpson/model/SimpsonCharacter.java (hunk 1 above)
Comment (aghigo) [reply src/main/java/com/javachallengers/simpson/model/SimpsonCharacter.java:19]: Done.

File: src/main/java/com/javachallengers/simpson/model/dto/ApiErrorDTO.java (hunk 2 above)
Comment (aghigo) [reply src/main/java/com/javachallengers/simpson/model/dto/ApiErrorDTO.java:14]: No, we don't. Changed from @Data to @Getter

File: src/main/java/com/javachallengers/simpson/model/dto/SimpsonCharacterRequestDTO.java (hunk 3 above)
Comment (aghigo) [reply src/main/java/com/javachallengers/simpson/model/dto/SimpsonCharacterRequestDTO.java:18]: No. Changed from @Data to @Getter

File: src/main/java/com/javachallengers/simpson/service/SimpsonCharacterService.java (hunk 4 above)
Comment (aghigo) [reply src/main/java/com/javachallengers/simpson/service/SimpsonCharacterService.java]: I didn't understand exactly. I refactored to use builder.

File: src/main/java/com/javachallengers/simpson/model/dto/ApiErrorDTO.java [hunk 5]
```diff
@@ -3,35 +3,16 @@
 import java.io.Serializable;
 import java.time.LocalDateTime;
 
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
```
Comment (rafadelnero) [inline src/main/java/com/javachallengers/simpson/model/dto/ApiErrorDTO.java:9]: Great, that's much better!

**Pull request merged**